### PR TITLE
CI: update hacl-packages workflow

### PR DESCRIPTION
**Pull request by pnmadelaine**
## Proposed changes

Update the `hacl-packages-create-branch` workflow:
- clearer comment on the PR
- don't fail on empty changes
- add link for the js bindings

## Types of changes

- [ ] Proof maintenance (non-breaking change which fixes a proof regression)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New algorithm or feature (non-breaking change which adds functionality)
- [ ] Improved performance (fix or feature that would improve performance of some algorithm on some platform)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [x] Continuous integration

## Checklist

- [x] I have read the [CODE_OF_CONDUCT.md](https://github.com/hacl-star/hacl-star/blob/main/CODE_OF_CONDUCT.md) doc
- [x] I have read and agree to submit my changes under the [LICENSE](https://github.com/hacl-star/hacl-star/blob/main/LICENSE)
- [x] I have added necessary documentation (if appropriate)
- [x] I have edited [CHANGES.md](https://github.com/hacl-star/hacl-star/blob/main/CHANGES.md) (if appropriate)

### CODE REVIEW

File: .github/workflows/hacl-packages-create-branch.yml [hunk 1]
```diff
@@ -34,14 +34,20 @@ jobs:
...
         uses: thollander/actions-comment-pull-request@v1
         with:
           message: |
-            [CI] Extensive testing of the generated code on hacl-packages:
-            - [build](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Abuild_pull_request)
-            - [ocaml](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Aocaml)
-            - [benchmarks](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Abenchmark_pull_request)
+            [CI] Important!
+            The code in `dist` is tested on [cryspen/hacl-packages](https://github.com/cryspen/hacl-packages).
+            `dist` is not automatically re-generated, be sure to do it manually.
```
Comment (franziskuskiefer) [inline .github/workflows/hacl-packages-create-branch.yml:45]: Let's add some line breaks where it makes sense in this paragraph to make it easier to read.

File: .github/workflows/hacl-packages-create-branch.yml [hunk 2]
```diff
@@ -34,14 +34,20 @@ jobs:
...
+            Then check the following tests before merging this PR.
+            Always check the latest run, it corresponds to the most recent version of this branch.
+            All jobs are expected to be successful.
+            In some cases manual intervention is needed. Please ping the hacl-packages maintainers.
+            - [Build & Test](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Abuild_pull_request): Build on Linux (x86, x64), Windows (x86, x64), MacOS (x64, arm64), s390x, Android (arm64) and test on Linux (x86, x64), Windows (x86, x64), MacOS (x64).
+            - [Benchmarks](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Abenchmark_pull_request): Navigate to the terminal output in “Run benchmarks”. The comparison with the main branch will be at the bottom.
+            - [OCaml bindings](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Aocaml) & [JS bindings](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Ajs)
```
Comment (franziskuskiefer) [inline .github/workflows/hacl-packages-create-branch.yml:52]: Maybe say that these build and test the bindings.

File: .github/workflows/hacl-packages-create-branch.yml [hunk 3]
```diff
@@ -34,14 +34,20 @@ jobs:
...
-            - [ocaml](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Aocaml)
-            - [benchmarks](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Abenchmark_pull_request)
+            [CI] Important!
+            The code in `dist` is tested on [cryspen/hacl-packages](https://github.com/cryspen/hacl-packages).
+            `dist` is not automatically re-generated, be sure to do it manually.
+            Then check the following tests before merging this PR.
+            Always check the latest run, it corresponds to the most recent version of this branch.
+            All jobs are expected to be successful.
+            In some cases manual intervention is needed. Please ping the hacl-packages maintainers.
+            - [Build & Test](https://github.com/cryspen/hacl-packages/actions?query=branch%3Ahacl-star-${{ github.head_ref }}+workflow%3Abuild_pull_request): Build on Linux (x86, x64), Windows (x86, x64), MacOS (x64, arm64), s390x, Android (arm64) and test on Linux (x86, x64), Windows (x86, x64), MacOS (x64).
```
Comment (franziskuskiefer) [inline .github/workflows/hacl-packages-create-branch.yml:50]: This also includes benchmarks (with different compilers etc.) but no comparison with main.
Add something to the description about the differences. The next bulletpoint is really about performance regression tests.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, a few more things, then this looks good to me.
Maybe @victor-dumitrescu  or @polubelova want to check as well to see if this makes sense to them.

File: .github/workflows/hacl-packages-create-branch.yml (hunk 1 above)
Comment (msprotz) [reply .github/workflows/hacl-packages-create-branch.yml:45]: `dist/gcc-compatible` and `dist/msvc-compatible`, I take it?

Comment (franziskuskiefer) [reply .github/workflows/hacl-packages-create-branch.yml:45]: Yes, and wasm.
Let's add thes to be more precise.

**Comment by franziskuskiefer** [comment]
@pnmadelaine can we get this updated and merged?

**Comment by victor-dumitrescu** [comment]
> 

Thanks, LGTM

File: .github/workflows/hacl-packages-create-branch.yml (hunk 1 above)
Comment (polubelova) [inline .github/workflows/hacl-packages-create-branch.yml:45]: > `dist` is not automatically re-generated, be sure to do it manually.

Should we say that a fresh dist for a successful build can be downloaded from CI (e.g., https://everest-ci.paris.inria.fr/build/7308)? So, there is no need to generate it locally.

**Review by franziskuskiefer — APPROVED** [review]
I added some of the suggestions.

**Review by R1kM — APPROVED** [review]
Thanks all, looks great. Now that there are more explanations, we should discuss whether it's worth reintroducing the CI check for the dist/gcc-compatible diff.
In the meantime, merging this

**Pull request merged**